### PR TITLE
Calentre-client-14-15: adding calendar + time selector components

### PR DESCRIPTION
**Pull request by luisflores93**
adding calendar + time selector components

### CODE REVIEW

File: prisma/schema.prisma
Comment (fiizzy) [inline prisma/schema.prisma:1]: Heyy @luisflores93 what does this file do?

File: src/components/Calendar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                );
+              onSelectDate(null);
+            }}
+            className={classNames(
+              'flex h-6 w-6 items-center justify-center rounded-full',
+              {
+                'bg-off-gray':
+                  selectedMonth === now.getMonth() &&
+                  selectedYear === now.getFullYear(),
+                'bg-[linear-gradient(180deg,_#A675E9_0%,_#0AAEF8_100%)]':
```
Comment (fiizzy) [inline src/components/Calendar/index.tsx:147]: Can we have the definitions of these color classes programatically?

```js
const colors = {
green: "xxxx",
blue: "xxxx",
}

```

File: src/components/Calendar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                  selectedDateTime.setMonth(selectedDateTime.getMonth() + 1)
+                )
+              );
+              onSelectDate(null);
+            }}
+            className={classNames(
+              'flex h-6 w-6 items-center justify-center rounded-full',
+              {
+                'bg-off-gray': false,
+                'bg-[linear-gradient(180deg,_#A675E9_0%,_#0AAEF8_100%)]': true,
```
Comment (fiizzy) [inline src/components/Calendar/index.tsx:171]: Same as above ^

File: src/components/Scheduler/AvailableTimes.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+export const AvailableTimes = ({
+  selectedDay,
+  dateFormat = 'hh:mm a',
+}: Props) => {
+  const { setSelectedHour, selectedHour } = useSchedulerContext();
+
+  return (
+    <div className='flex flex-col items-center gap-5'>
+      <p>{format(selectedDay.day, 'EEEE, MMM dd')}</p>
```
Comment (fiizzy) [inline src/components/Scheduler/AvailableTimes.tsx:20]: What does EEEE stand for in the date format?

File: src/components/Scheduler/AvailableTimes.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      <div className='flex h-full max-h-[270px] w-full min-w-[135px] flex-col items-center gap-[8px] overflow-auto'>
+        {selectedDay.hours.map((time, index) => (
+          <Button
+            className={classNames('py-1', {
+              'border-transparent bg-[linear-gradient(180deg,_#F8B453_0%,_#F8419D_100%)] text-white':
```
Comment (fiizzy) [inline src/components/Scheduler/AvailableTimes.tsx:25]: Same programatic color coding

File: src/components/Scheduler/DateDetails.tsx [hunk 5]
```diff
@@ -1,46 +1,73 @@
...
-  const toggleAmPm = (value: string) => {
-    setTogglerValue(value);
-  };
   return (
     <div className='border-l border-off-gray pl-7'>
       <div className='flex flex-col gap-6 py-10 pr-10'>
         <div className='flex items-start'>
           <p className='text-[22px] font-semibold leading-none tracking-[0.15px] text-gray'>
-            Select date and time
+            {translate('Select date and time')}
```
Comment (fiizzy) [inline src/components/Scheduler/DateDetails.tsx:27]: Support for multiple language?

File: src/helpers/getUserDetails.ts
Comment (fiizzy) [inline src/helpers/getUserDetails.ts:1]: Love how you handled this function! ✅

**Review by fiizzy — CHANGES_REQUESTED** [review]
Everything mostly looks good! Great job ✅ 

Just have a few reviews which I have commented. You can have a look, and if any is not clear, you can just easily reach out to me on Discord for more clarifications.

File: src/components/Scheduler/DateDetails.tsx (hunk 5 above)
Comment (luisflores93) [reply src/components/Scheduler/DateDetails.tsx:27]: > Support for multiple language?

yeah, I created this function for when we want to implement multiple languages

File: prisma/schema.prisma
Comment (luisflores93) [reply prisma/schema.prisma:1]: This file (short: schema file, Prisma schema or schema) is the main configuration file for your Prisma ORM setup.
ref: [docs](https://www.prisma.io/docs/orm/prisma-schema/overview)

File: src/components/Scheduler/AvailableTimes.tsx (hunk 3 above)
Comment (luisflores93) [reply src/components/Scheduler/AvailableTimes.tsx:20]: thats for the day of the week, you can found more format references here: [date-fns formats](https://date-fns.org/v3.3.1/docs/format)

Comment (fiizzy) [reply src/components/Scheduler/AvailableTimes.tsx:20]: Got it!

File: prisma/schema.prisma
Comment (fiizzy) [reply prisma/schema.prisma:1]: Awesome!
Thanks for the link.

**Comment by fiizzy** [comment]
LGTM

**Pull request merged**